presto-hive/src/main/java/com/facebook/presto/hive/HivePartitionManager.java

```diff
@@ -360,24 +364,6 @@ private boolean queryAccessesTooManyBuckets(HiveBucketHandle handle, Optional<Hi
         return bucketsPerPartition * partitions.size() > getMaxBucketsForGroupedExecution(session);
     }
 
-    private List<HivePartition> getPartitionsAsList(Iterator<HivePartition> partitionsIterator)
-    {
-        ImmutableList.Builder<HivePartition> partitionList = ImmutableList.builder();
-        int count = 0;
-        while (partitionsIterator.hasNext()) {
-            HivePartition partition = partitionsIterator.next();
-            if (count == maxPartitionsPerScan) {
-                throw new PrestoException(HIVE_EXCEEDED_PARTITION_LIMIT, format(
-                        "Query over table '%s' can potentially read more than %s partitions",
-                        partition.getTableName(),
-                        maxPartitionsPerScan));
-            }
-            partitionList.add(partition);
-            count++;
-        }
-        return partitionList.build();
-    }
-
```

Comment:
@rschlussel  was this change introduced to lazily get the next partition and keep track of the count and fail if it is above maxPartitionsPerScan ?.

Reply:
so, i didn't introduce this code i just moved it (original changes came from here https://github.com/prestodb/presto/pull/6751/commits/47ae18c25350708eb5b9b6d4eb3fb8b50a66c359 and https://github.com/prestodb/presto/pull/10259/commits/59dcbd22706ed2eef798bf9dbb9baa08509f2b5e).  But, yeah, I think it's basically there for enforcing the maxPartitionsPerScan limit without having to enumerate all the partitions first in case it's really big.

Reply:
Thanks Rebecca. since we are not really passing in a Iterator rather grabbing the Iterator from the Iterable List from line 191 we have the entire size already available to us and this PR refactoring makes sense.